launcher/launcher.go

```diff
@@ -627,110 +394,77 @@ func (launcher *Launcher) runInstances(runInstances []InstanceInfo) {
 	}()
 
 	launcher.runMutex.Lock()
-
 	launcher.runInstancesInProgress = true
-
-	if launcher.currentInstances == nil {
-		launcher.currentInstances = make(map[string]*instanceInfo)
-	}
+	launcher.runMutex.Unlock()
 
 	launcher.cacheCurrentServices(runInstances)
 
-	stopInstances := launcher.getStopInstances(runInstances)
-	startInstances := launcher.getStartInstances(runInstances)
-
-	launcher.runMutex.Unlock()
+	stopInstances, startInstances := launcher.calculateInstances(runInstances)
 
 	launcher.stopInstances(stopInstances)
 	launcher.startInstances(startInstances)
-
-	launcher.checkNewServices()
 }
 
-func (launcher *Launcher) checkNewServices() {
-	launcher.errorServices = nil
-
-serviceLoop:
-	for _, service := range launcher.currentServices {
-		if service.IsActive || errors.Is(service.err, servicemanager.ErrNotExist) {
-			continue
-		}
-
-		for _, instance := range launcher.currentInstances {
-			if instance.ServiceID == service.ServiceID &&
-				instance.runStatus.State == cloudprotocol.InstanceStateActive {
-				log.WithFields(log.Fields{"serviceID": service.ServiceID}).Debug("Apply new service")
+func (launcher *Launcher) calculateInstances(
+	runInstances []InstanceInfo,
+) (stopInstances, startInstances []*runtimeInstanceInfo) {
+	launcher.runMutex.Lock()
+	defer launcher.runMutex.Unlock()
 
-				if err := launcher.serviceProvider.ApplyService(service.ServiceInfo); err != nil {
-					log.WithField("serviceID", service.ServiceID).Errorf("Can't apply service: %v", err)
+	currentInstances := make([]*runtimeInstanceInfo, 0, len(launcher.currentInstances))
 
-					launcher.errorServices = append(launcher.errorServices,
-						service.cloudStatus(cloudprotocol.ErrorStatus, err))
-				}
+	for _, currentInstance := range launcher.currentInstances {
+		currentInstances = append(currentInstances, currentInstance)
```

Comment:
you could use `[]` instead `append`

Reply:
I can't use it directly. launcher.currentInstances is the map. 
```go
i := 0

for _, currentInstance := range launcher.currentInstances {
    currentInstances[i] = currentInstance
    i++ 
}
```

with append it is shorter.